lib/kubeclient/common.rb

```diff
@@ -337,19 +342,18 @@ def net_http_options(uri)
       }
 
       if uri.scheme == 'https'
-        options.merge!(
-          use_ssl: true,
+        options[:ssl] = {
```

Comment:
http.rb takes SSL options from an `ssl` sub hash under main options, see https://github.com/httprb/http/blob/1836ae8d7986d4ce39ab417bdfd20b9098800edc/lib/http/options.rb#L47

Furthermore, `use_ssl` is not required for http.rb, it's a net/http concept, see http://ruby-doc.org/stdlib-2.2.4/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-HTTPS